src/dev/lupio/lupio_rtc.cc

```diff
@@ -0,0 +1,103 @@
+#include "dev/lupio/lupio_rtc.hh"
+
+#include <ctime>
+#include <string>
+
+#include "base/bitfield.hh"
+#include "base/time.hh"
+#include "base/trace.hh"
+#include "debug/LupioRTC.hh"
+#include "mem/packet.hh"
+#include "mem/packet_access.hh"
+#include "params/LupioRTC.hh"
+#include "sim/system.hh"
+
+namespace gem5
+{
+
+using namespace RiscvISA;
+
+LupioRTC::LupioRTC(const Params &params) :
+    BasicPioDevice(params, params.pio_size),
+    system(params.system),
+    time(params.time)
+{
+    struct timeval tv;
+    struct tm time_bd;
+
+    gettimeofday(&tv, NULL);
+    gmtime_r(&tv.tv_sec, &time_bd);
+
+    DPRINTFN("Real-time clock set to %s", asctime(&time_bd));
```

Comment:
I'm not sure what this constructor is supposed to do. Both variables are local, so they will disappear when the function returns.

Reply:
Also on this line. Instead of `DPRINTFN`, it would be better to use `DPRINTF(LupioRTC, ...`